modules/cockpit/src/app/sendEventView/send.eventview.ts

```diff
@@ -53,53 +55,84 @@ export class SendEventViewComponent implements OnInit {
     }
 
     public sendEvent() {
-        if(!this.topicName){
-            return
-        }
-        this.loading = true;
+      if (!this.topicName) {
+        return;
+      }
+      let eventData;
+      if (!this.cloudevent) {
         let headers = new Headers({ 'Content-Type': 'application/json' });
-        let httpOptions = new RequestOptions({ headers: headers });
-        var editor = ace.edit("eventTopicEditor");
-        let eventTime = new Date().toISOString();
-        let eventType = this.topicName;
-        var version;
-        var regex = /^(.*)\.([v|V][0-9]+$)/;
-        if (eventType.match(regex)) {
-            var matchedGroups = regex.exec(eventType);
-            version = matchedGroups[2];
-            eventType = matchedGroups[1];
-        }
-        else {
-            version = this.event.events.spec.info.version;
-        }
-        try {
-            let eventData = {
-                "event-type": eventType,
-                "event-type-version": version, //event types normally end with .v1
-                "event-time": eventTime,
-                "data": JSON.parse(editor.getValue()),
-                "event-tracing": this.tracing
-            }
-            this.http.post(this.baseUrl + this.info.links.events, eventData, httpOptions)
-                .subscribe(
-                    success => {
-                        this.loading = false;
-                        this.successMessage = "Event has been sent Successfully";
-                        this.success = true;
-                    },
-                    error => {
-                        this.alertMessage = JSON.parse(error._body).error
-                        this.alert = true;
-                        this.loading = false;
-                    });
-        }
-        catch (err) {
-            this.alertMessage = "Please make sure that the Event Data follows JSON format";
-            this.alert = true;
-            this.loading = false;
-        }
+        var httpOptions = new RequestOptions({ headers });
+        eventData = this.sendLegacyEvent();
+      } else {
+        let headers = new Headers({ 'Content-Type': 'application/cloudevents+json' });
+        var httpOptions = new RequestOptions({ headers });
+        eventData = this.sendCloudEvent();
+      }
+      try {
+        this.http.post(this.baseUrl + this.info.links.events, eventData, httpOptions)
+            .subscribe(
+                success => {
+                  this.loading = false;
+                  this.successMessage = "Event has been sent Successfully";
+                  this.success = true;
+                },
+                error => {
+                    this.alertMessage = JSON.parse(error._body).error;
+                    this.alert = true;
+                    this.loading = false;
+                });
+      } catch (err) {
+        this.alertMessage = "Please make sure that the Event Data follows JSON format";

```

Comment:
That message is shown in the UI and will not help the end user... what can he do about it?

Reply:
Improved the message.